### PR TITLE
Updated Error Codes II

### PR DESCRIPTION
**Pull request by msaxena2**
This PR follows from #180 . The commits can be used to view the diff in manageable chunks. @grosu @dwightguth please review.

### CODE REVIEW

File: examples/simple/Error_Codes.csv [hunk 1]
```diff
@@ -48,109 +48,108 @@ UB-CEER1,Trying to read through a null pointer.,"6.3.2.1:1, J.2:1 item 19",Undef
...
+UB-FD5,Function main must return an int.,"5.1.2.2.1:1, J.2:1 item 4",Undefined Behavior
+UB-FD6,"If main has arguments, the type of the first argument must be equivalent to int.","5.1.2.2.1:1, J.2:1 item 4",Undefined Behavior
+UB-FD7,"If main has arguments, the type of the second argument must be equivalent to char**.","5.1.2.2.1:1, J.2:1 item 4",Undefined Behavior
+UB-FD8,Function main must have zero or two parameters.,"5.1.2.2.1:1, J.2:1 item 4",Undefined Behavior
+UB-FD9,Function main must have zero or two parameters.,"5.1.2.2.1:1, J.2:1 item 4",Undefined Behavior
 UB-CTI1,Arrays must have integer length.,"6.6:6, J.2:1 item 55",Undefined Behavior
-UB-STDARG1,"'va_start' used in function with fixed args.","6.9.1:8, J.2:1 item 87",Undefined Behavior
-UB-STDIO1,"printf: Format string expects type of size a, but got an argument of size b.","7.21.6.1:9, J.2:1 item 153",Undefined Behavior
+UB-STDARG1,'va_start' used in function with fixed args.,"6.9.1:8, J.2:1 item 87",Undefined Behavior
+UB-STDIO1,"'printf': Format string expects type of size a, but got an argument of size b.","7.21.6.1:9, J.2:1 item 153",Undefined Behavior
```
Comment (grosu) [inline examples/simple/Error_Codes.csv]: Are size "a" and size "b" correct here?  Are they replaced with actual sizes when the error message is reported?

File: examples/simple/Error_Codes.csv [hunk 2]
```diff
@@ -48,109 +48,108 @@ UB-CEER1,Trying to read through a null pointer.,"6.3.2.1:1, J.2:1 item 19",Undef
...
-CV-EFNC2,incompatible types in function call arguments.,6.5.16.1:1,Constraint Violation
-CV-EFNC3,incompatible pointer types in function call arguments.,6.5.16.1:1,Constraint Violation
-CV-TDG2,declaration for appears with multiple storage class specifiers.,6.7.1:2,Constraint Violation
-CV-TDG4,declaration appears with invalid storage class specifiers.,6.9:2,Constraint Violation
-CV-TDG5,function parameter appears with storage class other than register.,6.7.6.3:2,Constraint Violation
-CV-TDG8,declaration has initializer but incomplete type.,6.7.9:3,Constraint Violation
-CV-TDG10,conflicting types for identifier.,6.7:4,Constraint Violation
+CV-EFNC2,Incompatible types in function call arguments.,6.5.16.1:1,Constraint Violation
+CV-EFNC3,Incompatible pointer types in function call arguments.,6.5.16.1:1,Constraint Violation
+CV-TDG2,Declaration for appears with multiple storage class specifiers.,6.7.1:2,Constraint Violation
```
Comment (grosu) [inline examples/simple/Error_Codes.csv]: Declaration for WHAT?

File: examples/simple/Error_Codes.csv [hunk 3]
```diff
@@ -48,109 +48,108 @@ UB-CEER1,Trying to read through a null pointer.,"6.3.2.1:1, J.2:1 item 19",Undef
...
 CV-TDL2,Extern declaration appears at block scope with an initializer.,6.7.9:5,Constraint Violation
 CV-TDL3,Extern declaration with incompatible type,6.7:4,Constraint Violation
 CV-TDL4,Redeclaration with no linkage,6.7:3,Constraint Violation
 CV-CDI4,Attempting to initialize nonexistent member.,6.7.9:2,Constraint Violation
-CV-TDG14,redefinition of symbol with external linkage,"6.9:5""",Constraint Violation
-CV-TDG15,"redefinition of struct, union, or enum",6.7.2.3:1,Constraint Violation
-CV-TDG18,declaration for appears with invalid storage class specifiers.,6.7.1:3,Constraint Violation
-CV-TDG16,incompatible redeclaration of symbol with internal or external linkage,6.7:4,Constraint Violation
+CV-TDG15,"Redefinition of struct, union, or enum",6.7.2.3:1,Constraint Violation
+CV-TDG18,Declaration for appears with invalid storage class specifiers.,6.7.1:3,Constraint Violation
```
Comment (grosu) [inline examples/simple/Error_Codes.csv]: How is this different from CV-TDG2 ?

**Comment by grosu** [comment]
O could not spot any other problem besides the ones I reported.  I'm assuming that you have rerun all the examples and included the actual generated error mssages (as opposed to modifying the old ones by hand).  Thanks, @msaxena2!!!

**Comment by dwightguth** [comment]
looks good but can you please fix conflicts and double check that the changes Chris made in the PR I just merged get synced up with this convention?

@chathhorn just FYI we are going to have the convention of capitalized descriptions ending with a period now so please try to follow that in the future so that we don't get out of sync again.

**Comment by dwightguth** [comment]
Jenkins: test this please

File: examples/simple/Error_Codes.csv (hunk 1 above)
Comment (msaxena2) [inline examples/simple/Error_Codes.csv]: Yes, a and b are just placeholders.

File: examples/simple/Error_Codes.csv (hunk 3 above)
Comment (msaxena2) [inline examples/simple/Error_Codes.csv]: CV-TDG2 should occur when there is more than one storage class specifier in the identifier declaration, while TDG18 should occurs when the identifier declaration has an invalid storage class specifier. @dwightguth can you confirm if the intended usage in the semantics is such?

File: examples/simple/Error_Codes.csv (hunk 1 above)
Comment (grosu) [inline examples/simple/Error_Codes.csv]: @msaxena2 , my questions were rhetoric.  Other users may have similar doubts, so try to clarify the text.  For example, put in angle brackets paramters that will be instantiated in each error message.  Something like this, but maybe you can find something even better:

  Format string expects type of size <format_type_size>, but got an argument of size <actual_type_size>

Comment (msaxena2) [inline examples/simple/Error_Codes.csv]: Oops, sorry! Will make the text better.

**Comment by dwightguth** [comment]
This looks fine to me to be merged if you're done tweaking it.

**Comment by msaxena2** [comment]
I've merged this PR. The agreed upon convention for error code descriptions, from now on, is that the description must begin with an uppercase letter, and end with a period. Also, any new PR that modifies the error codes or descriptions will be expected to follow this convention.

**Comment by grosu** [comment]
Thanks.  Another question, also raised by @dwightguth , is to how make sure that we have tests for all the error codes, and that the .csv list is in synch with the code.  Not very urgent, but something to keep in mind and maybe come up with a policy for.

**Comment by msaxena2** [comment]
I've already opened an issue #183 to track this. @dwightguth has suggested some solutions.

**Pull request merged**